### ISSUE TITLE
Database cache and table cache timeout set to 0 or -1 crashes when creating dataset

### ISSUE THREAD

**Issue by larshelge** [apache/superset#25513]
Database connection cache and table cache timeout set to 0 or -1 crashes when attempting to create a dataset.

#### How to reproduce the bug

* Create a new Database connection. In my case of type Amazon Redshift.
* Go to step 3 and select Performance.
* Set Schema cache timeout to 0 seconds.
* Set Table cache timeout to 0 seconds.
* Save the Database connection.

* Go to Datasets in the top menu.
* Click + Dataset.
* Select the mentioned database.
* Select a schema.

This will throw an exception with the following text:

> There was an error loading the tables: Unexpected error occurred, please check your logs for details

If I set the cache timeouts to a positive integer, the tables load well.

Unfortunately the Docker Compose logs show nothing relevant.

### Expected results

I expect the tables for the schema to load, despite the cache timeout being set to 0.

*The larger question:* How do you disable the schema and table cache entirely? The UI does not indicate how to do that. Setting the value to 0 or -1 is my best guess.

### Actual results

Loading tables will throw an exception with the following text:

> There was an error loading the tables: Unexpected error occurred, please check your logs for details

Unfortunately the Docker Compose logs show nothing relevant.

#### Screenshots

![superset](https://github.com/apache/superset/assets/908206/f6f22845-1558-4421-ae0d-55b041e12f3a)

### Environment

- Chrome. Version 117.0.5938.92 (Official Build) (64-bit).
- Superset 3. 
- Docker Compose latest build, as of Oct 4, 2023.

### Additional info

The browser sends an API request:

> /api/v1/database/6/tables/?q=(force:!f,schema_name:public)

This fails with status `422 Unprocessable Content`.

**Comment by rusackas** [comment]
I don't have a redshift instance to test, but maybe @cbuffevant (from the [rolodex](https://docs.google.com/spreadsheets/d/1U1qxiLvOX0kBTUGME1AHHi6Ywel6ECF8xk_Qy-V9R8c/edit#gid=0), thanks for signing up!) might have some insight here.

**Comment by cbuffevant** [comment]
I was able to replicate it with a Redshift Serverless cluster, using Superset 3.0.2. I can take a deeper look...

**Comment by cbuffevant** [comment]
After removing the cache time out values, the error persists... 
![image](https://github.com/apache/superset/assets/8312839/fc96ff23-b1f4-4566-96ed-b4288f197fa8)

**Comment by rusackas** [comment]
Oh, interesting. Thanks for validating. I wonder if putting in a reasonable default clears the error? In other words are just 0/-1 problematic, but 60 or 600 are fine? Thanks for looking into this!

**Comment by cbuffevant** [comment]
Yes, I can confirm that setting timeout value to 1, works fine. The problem seems to be related with values <= 0.

**Comment by rusackas** [comment]
CCing @betodealmeida or @sadpandajoe in case they might be interested in this one.

**Comment by rusackas** [comment]
This has been silent for a long while now. Is it still an issue in 4.1.2/5.0.0?

**Comment by rusackas** [comment]
Closing this out as inactive/not planned, but MORE than happy to reopen/revisit if this is still an issue in 5.0.0/`master`
